asf_heat_pump_affordability/getters/get_data.py

```diff
@@ -1,20 +1,133 @@
 import requests
 import pandas as pd
+import numpy as np
 from io import BytesIO
+from zipfile import ZipFile
+from typing import Union
+from asf_heat_pump_affordability import config, json_schema
 
 
-def get_df_from_url(url: str) -> pd.DataFrame:
+def get_df_from_csv_url(url: str, **kwargs) -> pd.DataFrame:
     """
-    Get dataframe from .csv file stored at URL.
+    Get dataframe from CSV file stored at URL.
 
     Args
-        url (str): URL location of .csv file download
+        url (str): URL location of CSV file download
+        **kwargs for pandas.read_csv()
 
     Returns
-        pd.DataFrame
+        pd.DataFrame: dataframe from CSV file
+    """
+    content = _get_content_from_url(url)
+    df = pd.read_csv(content, **kwargs)
+
+    return df
+
+
+def get_df_from_excel_url(url: str, **kwargs) -> pd.DataFrame:
```

Comment:
I took a leaf from pandas here and separated the getters into different functions for different file types to avoid confusion/errors.